src/main/java/net/kodehawa/mantarobot/commands/utils/polls/Poll.java

```diff
@@ -0,0 +1,287 @@
+        if (getChannel() == null) {
+            ctx.edit("commands.poll.invalid_channel", EmoteReference.ERROR);
+            return;
+        }
+
+        var at = 0;
+        var user = ctx.getAuthor();
+        var languageContext = ctx.getLanguageContext();
+        var builder = new EmbedBuilder().setAuthor(String.format(languageContext.get("commands.poll.header"), user.getName()), null, user.getAvatarUrl())
+                .setColor(Color.CYAN)
+                .setThumbnail("https://i.imgur.com/7TITtHb.png")
+                .setFooter(String.format(languageContext.get("commands.poll.time"),
+                        Utils.formatDuration(languageContext, time() - System.currentTimeMillis())), user.getAvatarUrl()
+                );
+
+        for (var option : options()) {
+            if (option.length() >= 1024) {
+                ctx.edit("commands.poll.too_long", EmoteReference.ERROR);
+                return;
+            }
+
+            builder.addField("Option #%01d".formatted(++at), MarkdownSanitizer.sanitize(option), false);
+        }
+
+
+        var image = image();
+        if (image != null && EmbedBuilder.URL_PATTERN.asPredicate().test(image)) {
+            builder.setImage(image);
+        }
+
+        if (image != null && EmbedBuilder.URL_PATTERN.asPredicate().test(image)) {
+            builder.setImage(image);
+        }
+
+        var message = ctx.sendResult(builder.build());
+        var reactions = reactions(options().size());
+        for (String reaction : reactions) {
+            message.addReaction(Emoji.fromUnicode(reaction)).queue();
+        }
+
+        setMessageId(message.getId());
+        schedule();
+    }
+
+    public void end() {
+        // Cancel poll before doing existing checks -- we don't want the object to dangle.
+        cancel();
+
+        // Checks before sending the final poll message.
+        if (getGuild() == null)
+            return;
+
+        if (getChannel() == null)
+            return;
+
+        if (messageId() == null || messageId.isEmpty()) {
+            log.error("Null messageId on poll? This shouldn't happen, dump: {}", asJson().toString());
+            return;
+        }
+
+        getChannel().retrieveMessageById(messageId).queue(message -> {
+            var languageContext = new I18nContext(db.getGuild(getGuild()).getData(), null);
+            var user = message.getAuthor();
+            var embedBuilder = new EmbedBuilder()
+                    .setTitle(languageContext.get("commands.poll.result_header"))
+                    .setDescription(String.format(languageContext.get("commands.poll.result_screen"), user.getName(), name))
+                    .setFooter(languageContext.get("commands.poll.thank_note"), null);
+
+            var react = new AtomicInteger(0);
+            var counter = new AtomicInteger(0);
+
+            var votes = message.getReactions().stream()
+                    .filter(r -> react.getAndIncrement() <= options.size())
```

Comment:
What if someone adds a different reaction?

Reply:
Right, guess we're better off filtering by the number reactions... was there a regex for this?